File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    s = []
    t = []
    j = []
    k = []

    for i in range(1, denom1+1):
        if denom1 % i == 0:     
            s.append(i)

    for x in s:
        if denom2 % x == 0:
            t.append(x)

    d = denom1 * denom2 // t[-1]                        
    f = (d // denom1) * numer1 + (d // denom2) * numer2     
    
    for i in range(1,d+1):
        if d % i == 0:     
            j.append(i)

    for x in j:
        if f % x == 0:
            k.append(x)

    return [f / k[-1],d / k[-1]]

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.go
func solution(numer1 int, denom1 int, numer2 int, denom2 int) []int {
    answer := make([]int, 2)
        n3 := numer1 * denom2 + denom1 * numer2
        d3 := denom1 * denom2
        g := gcd(n3, d3)

        answer[0] = n3/g
        answer[1] = d3/g
        
        return answer
}

func gcd(a int, b int) int {
    if b == 0 {return a}

        return gcd(b, a % b)
    }